Centrality reco modification for 2018 heavy ion run

YeonjuGo opened the pull request:
fixed pixel collection name to be able to read them
and added hfecut, pixelplus, pixelminus, pixeltrackplus, pixeltrackminus variables
 : https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/181017_RecoModification_hf_pixel_variables_GO.pdf

updated pixel cut 
 :  https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/pixel-cluster-size-181018.pdf

YeonjuGo committed:
add hihfecut variable

YeonjuGo committed:
nPixel and nPixelTracks work now

YeonjuGo committed:
nPixelPlus and Minus added

YeonjuGo committed:
nPixelTracksPlus and Minus added

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24892/6858

cmsbuild commented:
A new Pull Request was created by  @YeonjuGo  for master.

It involves the following packages:

DataFormats/HeavyIonEvent
RecoHI/HiCentralityAlgos


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@yslai, @jazzitup, @yenjie, @rovere, @kurtejung, @mandrenguyen, @dgulhan, @yetkinyilmaz this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 7cb46bc84a8ec115d97fd98f6ae25f4631da9f08

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24892/31082/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24892/31082/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2994843
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2994645
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 134 log files, 14 edm output root files, 32 DQM output files

YeonjuGo commented:
@slava77 
Please take a look at the presentation summarized modifications and related plots; https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/181017_RecoModification_hf_pixel_variables_GO.pdf
Thank you

slava77 commented on RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc:
```diff
@@ -226,16 +228,20 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	      creco->etHFtowerSumMinus_ = inputCentrality->EtHFtowerSumMinus();
 	      creco->etHFtowerSumPlus_ = inputCentrality->EtHFtowerSumPlus();
+          creco->etHFtowerSumECutMinus_ = inputCentrality->EtHFtowerSumECutMinus();
+          creco->etHFtowerSumECutPlus_ = inputCentrality->EtHFtowerSumECutPlus();
```
please indent consistently

slava77 commented on RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc:
```diff
@@ -319,12 +331,19 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	  }
 	  nPixel++;
+      if(eta>=0) nPixel_plus++;
+      if(eta<0) nPixel_minus++;
```
indentation

slava77 commented on RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc:
```diff
@@ -414,11 +433,24 @@ CentralityProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+    int nPixelTracksPlus = 0;
+    int nPixelTracksMinus = 0;
+
+    for(unsigned int i = 0 ; i < pixeltracks->size(); ++i){
```
```for (auto const& track : pixelTracks)```

slava77 commented on RecoHI/HiCentralityAlgos/python/HiCentrality_cfi.py:
```diff
@@ -38,7 +38,8 @@
+               producePixelTracks = True,
+               srcPixelTracks = cms.InputTag("hiConformalPixelTracks"),
                srcTracks = cms.InputTag("generalTracks"),
                srcVertex = cms.InputTag("offlinePrimaryVertices")
```
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

YeonjuGo committed:
fixed indentation and class version etc. according to the PR comments and modified pixel cut

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24892/6915

YeonjuGo commented:
@slava77 
I fixed things as you commented and also updated one more part. 
The new update is about pixel cut. We've been using this cut which is tuned for pre-phase 1 pixel detectors. So we'd like to update it. You can refer to the study on this: https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2018/pixel-cluster-size-181018.pdf
Thank you.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 16baa76797d6f5b9a11ef61b2a142147b3ed9ac3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24892/31183/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24892/31183/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2994843
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2994644
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 134 log files, 14 edm output root files, 32 DQM output files

fabiocos commented:
+1
